R/hex-to-dec.R

```diff
@@ -20,6 +21,6 @@ hex_to_dec <- function(hex) {
   hex[is.na(hex)] <- "NaN"
   dec <- as.numeric(Rmpfr::mpfr(hex, base = 16))
   # switch NaN's back to regular NA's
-  dec[is.nan(dec)] <- NA
+  dec[is.nan(dec)] <- NA_real_
```

Comment:
should it be returning integers (and using NA_integer_) or is this not possible with the numbers because they are so big?

Reply:
You got it, the numbers are too big. When you set them to be integer it turns them to be NA

Reply:
I thought real was slightly better than character but not sure what the best is in a case like this.

Reply:
I'm not sure either - is there a chance of two pit tags looking the same but not matching because of a difference at a small number of decimal places?

Reply:
I haven't seen one convert to have a decimal number, the ones I have seen always convert to whole numbers. 

My understanding of a hex code is it will always convert to a whole number/integer, the issue we have is that R can't seem to handle these large integers. Each hex coding should always generate the same whole number.

If it appends a decimal in the background because the value in a binary representation actually has decimal points maybe this would cause an issue. I am not sure nor have I found an example that does that. 

I have tested the function against the set of values we have from one study and they all matched up perfectly in that case. 

If you feel character is `NA_character_` would be safer I can update to that.

Reply:
No I think we go with real and change to character if we encounter problems.